### ISSUE TITLE
Cluster starts misbehaving after a while

### ISSUE THREAD

**Issue by nicoekkart** [networkservicemesh/networkservicemesh#2161]
I've also asked this on cnf-testbed and they've redirected me here.

I'm working on GoEPC using nsm, and for this I've set up my own cluster on a virtualized testbed from my university. Everything works fine, and all use cases seem to work at first. After a while, some strange behaviour starts happening, and I have no idea what the root cause is of this. This might not be related to the cnf-testbed or nsm, but maybe someone here knows where to look. Whenever this happens, my current only option is to set up the cluster again, which costs me about an hour (which is not ideal).

**Setup**
My setup looks like this:
![image](https://user-images.githubusercontent.com/2086383/83251881-94babe00-a1aa-11ea-9fd4-a31844c44852.png)
4 nodes, a master and a controlling PC. From the controlling PC, I'm running an adapted version of `make k8s` and afterwards an adapted version of `make vswitch`. I've removed some tasks from these scripts, which didn't seem necessary for me. This adapted code can be found at [my fork](https://github.com/nicoekkart/cnf-testbed).

**Problem**
I'm constantly installing helm charts and uninstalling them (approx 1 full restart in 10 minutes). After maybe 3 hours of working on 1 cluster, suddenly one or two nodes start misbehaving. I first notice the misbehaving because of an **UnexpectedAdmissionError** of a pod on the problematic node. When looking at more detail from the kubelet, it says: `Update plugin resources failed due to no containers return in allocation response`. This looks like something from the device plugin. 
When then trying to delete the pods, the ones on that node are not deleted, but all containers in that pod are stopped. 

I have looked at possible OOM, file openers running out, ...
I'm running out of ideas where to look for the cause of this problem. The weird thing is that everything is always fine at the beginning, but it suddenly starts to crash. My idea is that something is filling up, but I have no idea what. I have not found a way to recover a crashed cluster, so I end up creating a new one all the time.

**Comment by edwarnicke** [comment]
Have you looked at your Node Resources to see if they are showing nsm socket devices in the expected way?

**Comment by nicoekkart** [comment]
While digging, I'm finding a lot of mentions to "too many open files". Is it normal for nsmd to have +600 fd's open? On my broken node, this is the case, on another node in this same cluster, there are 400+ fd's open. On a brand new cluster with NSM installed, 38 fd's are open for nsmd. 

Is it possible that there's some problem with closing files in nsmd (or anything else I should look at)?

**Comment by nicoekkart** [comment]
Yes! By removing all /var/lib/networkservicemesh folders, the problem seems to be resolved. Should these old client connections normally be deleted?

**Comment by nicoekkart** [comment]
Does anybody have any clue about the cause of this buildup in /var/lib/networkservicemesh. I have it working for my cluster completely, by removing this, but where could this be coming from?

**Comment by edwarnicke** [comment]
@nicoekkart It's likely related to issues of cleanup with the device plugin.  Do you ever see kubelet reallocate a nsm socket?
